staging/src/kubevirt.io/client-go/api/v1/schema.go

```diff
@@ -390,6 +393,13 @@ type GPU struct {
 	DeviceName string `json:"deviceName"`
 }
 
+//
+// +k8s:openapi-gen=true
+type Filesystem struct {
+	// Name is the device name
+	Name string `json:"name"`
```

Comment:
do we want to open this up to the possibility of the Filesystem struct being able to represent different filesystem types in the future similar to how we do disks?

For example, i don't know if 9p would ever be supported, but if we ever did want to support another filesystem type we could extend the api pretty easily if we adopted a pattern like this.

```
type Filesystem struct {
    Name string `json:"name"`
    // pick one
    Virtiofs *FilesystemVirtiofs `json:"virtiofs"`
    9p *Filesystem9p `json:"9p"`

}
```

Reply:
I'm not against it, however, the reason we are introducing virtio-fs is that there are issues with 9p.

Reply:
yeah i agree about 9p. i guess structuring it this way leaves us open to alternatives in the future, maybe even a new emerging technology that doesn't exist right now.

Reply:
Yep, I agree. Added. Thanks.